workspaces/global-floating-action-button/plugins/global-floating-action-button/src/types.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export enum Position {
+  bottomRight = 'bottomRight',
+  bottomCenter = 'bottomCenter',
+}
+
+export type FloatingActionButtonWithPositions = Record<
+  Position | string,
+  FloatingActionButton[]
+>;
+
+export type FloatingActionButton = {
+  label: string;
+  showLabel?: boolean;
+  icon: React.ReactElement;
+  size?: 'small' | 'medium' | 'large' | string;
+  position?: Position | string;
+  color?:
+    | 'default'
+    | 'error'
+    | 'info'
+    | 'inherit'
+    | 'primary'
+    | 'secondary'
+    | 'success'
+    | 'warning'
+    | string;
+  onClick?: React.MouseEventHandler;
+  url?: string;
```

Comment:
We should also support links within the backstage instance like `/settings`. That's why I would suggest `to` here.

The Backstage `Link` component for example automatically detects external links and open them in a new page.

See also: https://github.com/backstage/backstage/blob/master/packages/core-components/src/components/Link/Link.tsx